### DISCUSSION TITLE
ios build crashes on launch when installed through test flight or .ipa file

### DISCUSSION THREAD

**Discussion by meehjeevan** [axmolengine/axmol#2174] [Q&A]
It's working fine when running from xcode to connected device but when it's installed through test flight or direct .ipa file it crashes on launch. When archiving debugsymbol is also not generated. So, I'm not able to symbolicate the crash log from ips file. 
Can anyone help?

**Comment by rh101** [comment]
You really should show your `CMakeLists.txt` (remove any sensitive info from it), otherwise it would just be guesses at what the problem may be.  The most important part would be the `if(APPLE)` section at the bottom of the default `CMakeLists.txt`, where all the properties are set for iOS etc..

**Reply by meehjeevan** [reply]
[CMakeLists.txt](https://github.com/user-attachments/files/17100717/CMakeLists.txt)
This is the "CMakeLists.txt" .

**Reply by rh101** [reply]
Try changing this line:
`set_target_properties(${APP_NAME} PROPERTIES RESOURCE "${APP_UI_RES}")`
to this:
```
set_target_properties(${APP_NAME} PROPERTIES
    RESOURCE "${APP_UI_RES}"
    XCODE_ATTRIBUTE_DEBUG_INFORMATION_FORMAT "dwarf-with-dsym"
    XCODE_ATTRIBUTE_GCC_GENERATE_DEBUGGING_SYMBOLS "YES"
    XCODE_ATTRIBUTE_ENABLE_STDEBUG_INFORMATION_FORMAT "dwarf-with-dsym"
    XCODE_ATTRIBUTE_DEPLOYMENT_POSTPROCESSING "YES"
)
```

Open up the generated package once it has finished building, and locate the dsym file.  If it's not there, then something else may be set incorrectly, such as a path etc... Do you by any chance set `XCODE_ATTRIBUTE_CONFIGURATION_BUILD_DIR` in any other `CMakeLists.txt` file?

**Comment by halx99** [comment]
Edit app build schema, change Archive settings to RelWithDebInfo may also generate debug info when archive and export .ipa

**Reply by meehjeevan** [reply]
still, dSYMs folder is empty but I got the crash logs in readable format now. I found the code where it exactly crashed. 
No I haven't set `XCODE_ATTRIBUTE_CONFIGURATION_BUILD_DIR` in any other `CmakeLists.txt` file.
